routing/dht/records.go

```diff
@@ -117,12 +124,19 @@ func (dht *IpfsDHT) verifyRecordLocally(r *pb.Record) error {
 // massive amplification attack on the dht. Use with care.
 func (dht *IpfsDHT) verifyRecordOnline(ctx context.Context, r *pb.Record) error {
 
-	// get the public key, search for it if necessary.
-	p := peer.ID(r.GetAuthor())
-	pk, err := dht.getPublicKeyOnline(ctx, p)
-	if err != nil {
-		return err
+	if len(r.Signature) > 0 {
+		// get the public key, search for it if necessary.
+		p := peer.ID(r.GetAuthor())
+		pk, err := dht.getPublicKeyOnline(ctx, p)
+		if err != nil {
+			return err
+		}
+
+		err = record.CheckRecordSig(r, pk)
+		if err != nil {
+			return err
+		}
```

Comment:
We can probably have some validators be pure and some not-- say, validation using `time` or network time is not pure (though Go may make it seem like it is).

i'm leaning towards sinking this into validators becuause it keeps the dht way cleaner, and makes them usable by other routing systems (am specifically thinking about mdns here)